lib/src/over_react_test/react_util.dart

```diff
@@ -481,8 +481,35 @@ Element queryByTestId(dynamic root, String value, {String key = defaultTestIdKey
 ///     </div>
 ///
 ///     queryAllByTestId(renderedInstance, 'value'); // returns both `inner` `<div>`s
-List<Element> queryAllByTestId(dynamic root, String value, {String key = defaultTestIdKey}) {
-  return findDomNode(root).querySelectorAll('[$key~="$value"]');
+List<Element> queryAllByTestId(dynamic root, String value, {String key = defaultTestIdKey, bool searchInShadowDom = false, int shadowDepth}) {
+  return _findAllDeep(findDomNode(root), _makeTestIdSelector(value, key: key), searchInShadowDom: searchInShadowDom, findMany: true, depth: shadowDepth);
+}
+
+String _makeTestIdSelector(String value, {String key = defaultTestIdKey}) => '[$key~="$value"]';
+
+dynamic /* Element | List<Element> */ _findAllDeep(Element root, String itemSelector, {bool searchInShadowDom = false, bool findMany = true, int depth}) {
+  List<Element> nodes = [];
+  var currentDepth = 0;
+  var lightElement = root.querySelector(itemSelector);
+  if (!findMany && lightElement != null){
+    return lightElement;
+  }
+  void recursiveSeek(dynamic _root) {
```

Comment:
I mean for most of its use its input is a `ShadowRoot`, but for the first input its `Element`, and the closest common parent class is `Node` but it doesn't have `querySelectorAll`
I could probably do like 
```
void recursiveSeek({ShadowRoot _shadowRoot, Element _element}) {
var _root = _shadowRoot ?? _element;
``` or something gross like that... idk any ideas?

Reply:
Mm that's fair; if there's no common ancestor of Element/ShadowRoot that has that shared API, `dynamic` is fine with me. 

Otherwise (typing it as `Node` or `Object` or something), the best I can think of would be:
```dart
void recursiveSeek(Node _root) {
  final rootQuerySelectorAll = root is ShadowRoot 
      ? root.querySelectorAll
      : root is Element ? root.querySelectorAll : null;

  nodes.addAll(rootQuerySelectorAll(itemSelector));

  ...
  var foundShadows = rootQuerySelectorAll(...
```